crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -2729,7 +2732,25 @@ mod tests {
         .expect("date out of range")
     }
 
-    fn call_member_state_event(
+    fn legacy_membership_for_my_call(
+        device_id: &str,
```

Comment:
Can't you use a `DeviceId` type from Ruma?

Reply:
`LegacyMembershipDataInit` should then definitly also require a DeviceId. This cannot be done in this PR but I will add the change to another PR.

Reply:
I added into this PR: https://github.com/ruma/ruma/pull/1908
It seems fitting to use this as the ruma cleanup PR.